README.md

```diff
@@ -14,169 +12,202 @@
-- [Vanilla Spark API](https://github.com/SnappyDataInc/snappy-poc/blob/master/src/main/scala/io/snappydata/adanalytics/SparkLogAggregator.scala) (from the original blog).
-- [Spark API with Snappy extensions](https://github.com/SnappyDataInc/snappy-poc/blob/master/src/main/scala/io/snappydata/adanalytics/SnappyAPILogAggregator.scala) to work with the stream as a sequence of DataFrames. (btw, SQL based access to streams is also the theme behind [Structured streaming](https://issues.apache.org/jira/browse/SPARK-8360) being introduced in Spark 2.0 )
-- [SQL based](https://github.com/SnappyDataInc/snappy-poc/blob/master/src/main/scala/io/snappydata/adanalytics/SnappySQLLogAggregator.scala) - described below.
-
-#### Generating the AdImpression logs 
-A [KafkaAdImpressionGenerator](src/main/scala/io/snappydata/adanalytics/KafkaAdImpressionProducer.scala) simulates Adservers and generates random [AdImpressionLogs](src/avro/adimpressionlog.avsc)(Avro formatted objects) in batches to Kafka.
-  ```scala
+We implemented the ingestion logic using [Vanilla Spark Structured Streaming](src/main/scala/io/snappydata/adanalytics/SparkLogAggregator.scala)
+and [Spark Structured Streaming with Snappy Sink](src/main/scala/io/snappydata/adanalytics/SnappyLogAggregator.scala)
+to work with the stream as a sequence of DataFrames.
+
+#### Generating the AdImpression logs    
+A [KafkaAdImpressionProducer](src/main/scala/io/snappydata/adanalytics/KafkaAdImpressionProducer.scala) simulates
+Adservers and generates random [AdImpressionLogs](src/avro/adimpressionlog.avsc)(Avro formatted objects) in batches to Kafka.
+
+```scala 
   val props = new Properties()
-  props.put("serializer.class", "io.snappydata.adanalytics.AdImpressionLogAvroEncoder")
-  props.put("partitioner.class", "kafka.producer.DefaultPartitioner")
-  props.put("key.serializer.class", "kafka.serializer.StringEncoder")
-  props.put("metadata.broker.list", brokerList)
-  val config = new ProducerConfig(props)
-  val producer = new Producer[String, AdImpressionLog](config)
-  sendToKafka(generateAdImpression())
-
-  def generateAdImpression(): AdImpressionLog = {
-    val random = new Random()
-    val timestamp = System.currentTimeMillis()
-    val publisher = Publishers(random.nextInt(NumPublishers))
-    val advertiser = Advertisers(random.nextInt(NumAdvertisers))
-    val website = s"website_${random.nextInt(Constants.NumWebsites)}.com"
-    val cookie = s"cookie_${random.nextInt(Constants.NumCookies)}"
-    val geo = Geos(random.nextInt(Geos.size))
-    val bid = math.abs(random.nextDouble()) % 1
-    val log = new AdImpressionLog()
-  }
-  
-  def sendToKafka(log: AdImpressionLog) = {
-    producer.send(new KeyedMessage[String, AdImpressionLog](
-      Constants.kafkaTopic, log.getTimestamp.toString, log))
+  props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer")
```

Comment:
Would be better if comments are mentioned before every code block / piece mentioning what it does.

Reply:
Added the comments.